tests/core/test_core_rpc_handlers.py

```diff
@@ -45,6 +54,74 @@ async def test_rpc_nodeInfo(make_request, enr):
     assert node_info["enr"] == repr(enr)
 
 
+@pytest.mark.trio
+async def test_rpc_updateNodeInfo(make_request, enr_manager):
+    # add a kv pair
+    first_response = await make_request("discv5_updateNodeInfo", [("0xabcd", "0x1234")])
+    first_enr = ENR.from_repr(first_response["enr"])
+    assert first_enr._kv_pairs[b"\xab\xcd"] == b"\x124"
+    assert first_enr.sequence_number == 2
+
+    # update a kv pair
+    second_response = await make_request(
+        "discv5_updateNodeInfo", [("0xabcd", "0x6789")]
+    )
+    second_enr = ENR.from_repr(second_response["enr"])
+    assert second_enr._kv_pairs[b"\xab\xcd"] == b"g\x89"
+    assert second_enr.sequence_number == 3
+
+    # test with multiple kv_pairs, and remove 'foo' key
+    third_response = await make_request(
+        "discv5_updateNodeInfo", [("0xabcd", None), ("0xdef1", 123)]
+    )
+    third_enr = ENR.from_repr(third_response["enr"])
+    assert b"\xab\xcd" not in third_enr._kv_pairs
+    assert third_enr._kv_pairs[b"\xde\xf1"] == to_bytes(123)
+    assert third_enr.sequence_number == 4
```

Comment:
Probably worth having at least one assertion against the `ENRManager.enr` itself to see that the value was actually updated on the other side of the API.  This is mostly just a sanity check.